### PR TITLE
[DPE-4264] backup test parity with vm charm

### PR DESCRIPTION
**Pull request by MiaAltieri**
## Issue 
Integration tests are missing latest improvements that have been made on VM charm

## Solution
Add them in

## Future
Lets consider make these into a package

### CODE REVIEW

**Review by Mehdi-Bendriss — DISMISSED** [review]

File: tests/integration/ha_tests/application_charm/src/continuous_writes.py
Comment (Gu1nness) [inline tests/integration/ha_tests/application_charm/src/continuous_writes.py:1]: This might fix https://warthogs.atlassian.net/browse/DPE-5002

**Review by Gu1nness — DISMISSED** [review]

File: tests/integration/ha_tests/application_charm/src/continuous_writes.py
Comment (MiaAltieri) [reply tests/integration/ha_tests/application_charm/src/continuous_writes.py:1]: Ah good eyes. Lets see if it does, 👀

File: tests/integration/backup_tests/helpers.py [hunk 1]
```diff
@@ -99,3 +101,14 @@ async def get_backup_list(ops_test: OpsTest, db_app_name=None) -> str:
     list_result = await action.wait()
     list_result = list_result.results["backups"]
     return list_result
+
+
+async def insert_unwanted_data(ops_test: OpsTest, app_name: str) -> None:
```
Comment (MiaAltieri) [inline tests/integration/backup_tests/helpers.py:106]: from VM charm

File: tests/integration/backup_tests/test_backups.py [hunk 2]
```diff
@@ -298,10 +290,10 @@ async def test_multi_backup(ops_test: OpsTest, github_secrets, continuous_writes
 
 @pytest.mark.group(1)
 @pytest.mark.abort_on_fail
-async def test_restore(ops_test: OpsTest, continuous_writes_to_db) -> None:
+async def test_restore(ops_test: OpsTest, add_writes_to_db) -> None:
```
Comment (MiaAltieri) [inline tests/integration/backup_tests/test_backups.py:293]: changes in this test are inline with VM charm

File: tests/integration/backup_tests/test_backups.py [hunk 3]
```diff
@@ -355,14 +343,10 @@ async def test_restore(ops_test: OpsTest, continuous_writes_to_db) -> None:
...
 
-# TODO remove unstable mark once juju issue with secrets is resolved
-
-
 @pytest.mark.group(1)
-@pytest.mark.unstable
 @pytest.mark.parametrize("cloud_provider", ["AWS", "GCP"])
 async def test_restore_new_cluster(
-    ops_test: OpsTest, github_secrets, continuous_writes_to_db, cloud_provider
+    ops_test: OpsTest, github_secrets, add_writes_to_db, cloud_provider
```
Comment (MiaAltieri) [inline tests/integration/backup_tests/test_backups.py:349]: changes in this test are inline with VM charm

File: tests/integration/backup_tests/test_sharding_backups.py [hunk 4]
```diff
@@ -12,7 +12,7 @@
 from tenacity import Retrying, stop_after_delay, wait_fixed
 
 from ..backup_tests import helpers as backup_helpers
-from ..ha_tests.helpers import deploy_and_scale_application, get_mongo_client
+from ..ha_tests.helpers import deploy_and_scale_application, get_direct_mongo_client
```
Comment (MiaAltieri) [inline tests/integration/backup_tests/test_sharding_backups.py:15]: update usage of client

File: tests/integration/ha_tests/helpers.py [hunk 5]
```diff
@@ -328,8 +328,8 @@ async def get_replica_set_primary(
     use_subprocess_to_get_password=False,
 ) -> Optional[Unit]:
     """Returns the primary unit name based no the replica set host."""
```
Comment (MiaAltieri) [inline tests/integration/ha_tests/helpers.py:330]: client usage in tests is messy, consolidate it

File: tests/integration/helpers.py [hunk 6]
```diff
@@ -652,9 +652,15 @@ async def wait_for_mongodb_units_blocked(
 
     This is necessary because the MongoDB app can report a different status than the units.
     """
-    for attempt in Retrying(stop=stop_after_delay(timeout), wait=wait_fixed(1), reraise=True):
-        with attempt:
-            await check_all_units_blocked_with_status(ops_test, db_app_name, status)
+    hook_interval_key = "update-status-hook-interval"
```
Comment (MiaAltieri) [inline tests/integration/helpers.py:655]: porting over change from VM

Comment (Gu1nness) [reply tests/integration/helpers.py:655]: When I have the time, I need to send the PR on pytest-operator to have this directly in the plugin code

**Review by Gu1nness — APPROVED** [review]

**Review by Mehdi-Bendriss — APPROVED** [review]

**Pull request merged**